### ISSUE TITLE
[GitHub Checks] Unblock GitHub checks when the failing tests are unrelated

### ISSUE THREAD

**Issue by KyleJu** [web-platform-tests/wpt-metadata#638]
Due to the increasing usage, GitHub checks have been failing >2 a week. The painpoints are caused by

- merge conflict, when multiple PRs triage the same files and the order of metadata entries change
- checks are blocked when unrelated tests are renamed, etc.

Per various offline discussions, the first issue can be improved by sorting the metadata entries but not completely eliminate the issue. To fix the second issue, we first need a reliable cron job to detect renaming/deletion before we unblock the check, because renaming/deletion will go undetected and require greater cleanup effort over time.

**Comment by KyleJu** [comment]
Duplicate of https://github.com/web-platform-tests/wpt-metadata/issues/639
